src/model/model.nit

```diff
@@ -1188,6 +1248,287 @@ abstract class MType
 	end
 end
 
+# A type that represents an operation over a set of types.
+abstract class MTypeSet[E: MType]
+	super MType
+
+	# The module of reference.
+	#
+	# Used to populate `model`, find primitive types and internally build new
+	# types with the same operator.
+	#
+	# SEE: `apply_to`
+	var mmodule: MModule
```

Comment:
* When comparing types and when adding `Object`.
* `apply_to` does use this attribute in all its concrete implementations.